Source/CombatExtended/CombatExtended/Comps/CompMeleeTargettingGizmo.cs

```diff
@@ -0,0 +1,254 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Verse;
+using RimWorld;
+using UnityEngine;
+
+namespace CombatExtended
+{
+    [StaticConstructorOnStartup]
+    public class MeleeTargettingAdd
+    {
+        static MeleeTargettingAdd()
+        {
+            foreach(ThingDef humanlike in DefDatabase<ThingDef>.AllDefs.Where(y => y.race != null && y.race.Humanlike))
+            {
+                if (humanlike.comps == null)
+                {
+                    humanlike.comps = new List<CompProperties>();
+                }
+
+                humanlike.comps.Add(new CompProperties { compClass = typeof(CompMeleeTargettingGizmo) });
+            }
+        }
+    }
+    public class CompMeleeTargettingGizmo : ThingComp
+    {
+        #region Fields
+        public Pawn PawnParent => (Pawn)this.parent;
+
+        public BodyPartHeight heightInt = BodyPartHeight.Undefined;
+
+        public BodyPartDef targetBodyPart = BodyPartDefOf.Torso;
+
+        #endregion
+
+        #region Properties
+
+        public Thing primaryWeapon
+        {
+            get
+            {
+                Thing result = null;

```

Comment:
This likely *shouldn't* be cached, as the weapon used can change rather often.  So you create a local variable, assign to it, then return it.  

This should just be `return PawnParent?.equipment.Primary;`